gslib/utils/cat_helper.py

```diff
@@ -136,6 +136,10 @@ def CatUrlStrings(self,
               print('==> %s <==' % blr)
               printed_one = True
             cat_object = blr.root_object
+            # This if statement ensures nothing is ouputted and no error
+            # is thrown if the user enters an out of bounds range for the object.
+            if start_byte >= cat_object.size:
+              return 0
```

Comment:
There's a `finally` clause here which sets `sys.stdout` to `old_stdout`, which would typically run before returning. I wonder if that could be related to some of the timeouts you're seeing.

Reply:
Nice catch!

Reply:
Wait, I think resetting `sys.stdout` isn't necessary:
```
def f():
    try:
        return 'returning'
    finally:
        print('finally')

print(f())

-> finally
-> returning
```

Reply:
Done.